synctl

```diff
@@ -861,6 +861,60 @@ class SyntheticPoP(Base):
                 f"Failed to get locations, status code {retrieve_res.status_code}")
             sys.exit(-1)
 
+    def get_location_summary_list(self):
+        """ --request POST 'http://{host}/api/synthetics/results/locationsummarylist'
+        --header "Authorization: apiToken <YourToken>" -i
+        --header "Content-Type: application/json"
+        -d '{
+             "pagincurlation": {
+             "page": 1,
+             "pageSize": 200
+            },
+             "timeFrame": {
+             "to": 0,
+             "windowSize": 3600000
+            }
+         }'
+         """
+
+        host = self.auth["host"]
+        token = self.auth["token"]
+        self.check_host_and_token(host, token)
+
+        headers = {
+            'Content-Type': 'application/json',
+            "Authorization": f"apiToken {token}"
+        }
+
+        summary_config = {
+            "pagination": {
+                "page": 1,
+                "pageSize": 200
+            },
+            "timeFrame": {
+                "to": 0,
+                "windowSize": 3600000
+            }
+        }
+        request_url = f"{host}/api/synthetics/results/locationsummarylist"
+        retrieve_res = requests.post(request_url,
+                                     headers=headers,
+                                     data=json.dumps(summary_config),
+                                     timeout=60,
+                                     verify=self.insecure)
+
+        if retrieve_res.status_code == 200:
+            data = retrieve_res.json()
+
+            if isinstance(data, dict):
+                return [data]
+            else:
+                return data
+        else:
+            return []
+
```

Comment:
is there any case returned data is not a dict? we can use it directly and return None if status code is not 200. turn it to list increase the complexity to handle it.

Reply:
Show an error message when status code is not 200.